hathor/p2p/manager.py

```diff
@@ -560,3 +596,114 @@ def drop_connection_by_peer_id(self, peer_id: str) -> None:
         protocol = self.connected_peers.get(peer_id)
         if protocol:
             self.drop_connection(protocol)
+
+    def sync_update(self) -> None:
+        """Update the subset of connections that running the sync algorithm."""
+        try:
+            self._sync_update_cmds()
+        except Exception:
+            self.log.error('_sync_update_cmds failed', exc_info=True)
+
+        try:
+            self._sync_rotate_if_needed()
+        except Exception:
+            self.log.error('_sync_rotate_if_needed failed', exc_info=True)
+
+    def _sync_update_cmds(self) -> None:
+        """Run sync_update commands. The supported commands are:
+
+        1. `always_enable_sync.txt` should contain a list of peer ids that will always have sync enabled.
+           This update skips lines starting with '#'.
+        """
+        cmd_path = self.manager.get_cmd_path()
+        if cmd_path is None:
+            return
+
+        always_sync_path = os.path.join(cmd_path, 'always_enable_sync.txt')
+        if not os.path.isfile(always_sync_path):
+            return
+
+        new = set()
+        with open(always_sync_path, 'r') as fp:
+            for line in fp.readlines():
+                line = line.strip()
+                if not line:
+                    continue
+                if line.startswith('#'):
+                    continue
+                new.add(line)
+
+        old = self.always_enable_sync
+        if new == old:
+            return
+
+        to_enable = new - old
+        to_disable = old - new
+
+        self.log.info('update always_enable_sync', new=new, to_enable=to_enable, to_disable=to_disable)
+
+        for peer_id in new:
+            if peer_id not in self.connected_peers:
+                continue
+            self.connected_peers[peer_id].enable_sync()
+
+        for peer_id in to_disable:
+            if peer_id not in self.connected_peers:
+                continue
```

Comment:
Maybe add a warning log in the `continue`? It's not an error, but probably indicates a human mistake when setting the cmd file

Reply:
Not really. We might add a list of predefined peer-ids where we are not connected to all of them. One can easily check the status of a peer using the HTTP API `/v1a/status`.